ulib/FStar.List.Tot.Properties.fst

```diff
@@ -258,6 +268,13 @@ val rev_involutive : l:list 'a ->
         (ensures (rev (rev l) == l))
 let rev_involutive l = rev_rev' l; rev_rev' (rev' l); rev'_involutive l
 
+(** Properties about snoc *)
+
+val snoc_length : l:list 'a -> x:'a ->
+  Lemma (requires True)
+        (ensures (length (snoc l x) = length l + 1))
+        [SMTPat (length (snoc l x))]
```

Comment:
This smt pattern should be removed. `snoc` is defined in terms of `append` and `FStar.List.Tot.Properties.append_length` already has an SMT pattern.

Reply:
Makes sense, removing.

Reply:
Just realized now why I had kept it: For some reason, the pattern on `FStar.List.Tot.Properties.append_length` doesn't seem to want to trigger for `snoc` invocations. Not entirely sure why. I'm keeping it for now, until I can analyze it further.

Reply:
I suspect that it is firing, but that you may be in a setting where you are working with `fuel = 0` or something like that. Because, with 0 fuel, `length [x] = 1` is not provable with SMT. That is provable with the normalizer though.

Reply:
Removed it. When I pushed the curried version of `snoc`, I had to update some of the other lemmas to ensure that the checks pass. 

I believe we should add in a `length (fst (unsoc l)) = length l - 1` lemma though, with a pattern on `length (fst (unsnoc l))`. Previously, the places that used `unsnoc` depended on unsnoc being related over to snoc, which then gave it the length. Now, the unsnoc/snoc relation is not directly triggered via simply using unsnoc and length, and thus it makes anything with an unsoc'd list (for example, an `index` operation) much more cumbersome. What do you think?